### PR TITLE
Fix #442: support globbing in file names and in list files

### LINKED ISSUE #442: z80asm: support globbing in command line and list files
See #440

### PR DESCRIPTION
**Pull request by pauloscustodio**
Add globbing support to z80asm in the command line args (for the Windows users) and
in list files. The standard '*' and '?' of glob.h are supported.

Added also the extended pattern '**' that expands to all subdirectories from the given point,
i.e. lib/**/*.asm will expand to all asm files in any subdirectory of lib.

### CODE REVIEW

**Comment by suborb** [comment]
Nice! I’ll take a look at this tomorrow night and make sure it works on a Mac.

**Comment by pauloscustodio** [comment]
I still want to test this in Ubuntu before merging.

I'm also planning to move the UNIXem submodule into an ext directory, to remove the clutter of the top level directory.

File: src/z80asm/options.c [hunk 1]
```diff
@@ -292,53 +297,158 @@ static char *search_source(char *filename)
...
+		leading[p - filename + 1] = '\0';
+		expand_glob(leading, sub_files);		// get my subdirs
+
+		// recurse for each directory (ends in /)
+		char **q = NULL;
+		while ((q = (char**)utarray_next(sub_files, q))) {
+			char *found = *q;
+			size_t len = strlen(found);
+
+			if (len > 0 && found[len - 1] == '/') {		// it's a directory, see GLOB_MARK
```
[the remark this answers is not in the document]
Comment (pauloscustodio) [reply src/z80asm/options.c]: This path delimiter is returned by glob() at the end of directories if the GLOB_MARK option is passed in.

The code was tested in Windows with MinGW and MSVC.

**Comment by pauloscustodio** [comment]
> It looks like we need to exclude directories from the glob result set otherwise we end up with errors like:
> Error at file 'x.lst' line 1: cannot read file 'obj/r2k/string/c//sccz80/'

I'm getting the same errors in Ubuntu.

File: src/z80asm/Makefile [hunk 2]
```diff
@@ -38,6 +38,11 @@ STD = -std=gnu11
 LOCAL_LIB :=	lib
 LOCAL_CFLAGS +=	-MMD -I. -I$(LOCAL_LIB) -It -g -Wall -ftabstop=4 $(STD) $(OPT)
 
+ifeq ($(OS),Windows_NT)
```
[the remark this answers is not in the document]
Comment (pauloscustodio) [reply src/z80asm/Makefile]: Yes, it is ugly... I'll put in a comment.

**Comment by pauloscustodio** [comment]
@suborb, I have now tested under MinGW, MSVC  and Ubuntu and fixed the errors you mentioned.
Can you please have a run in OSX before I merge?

**Pull request closed without merging**